quill-jdbc-zio/src/main/scala/io/getquill/context/qzio/ZioJdbcUnderlyingContext.scala

```diff
@@ -188,37 +178,58 @@ abstract class ZioJdbcUnderlyingContext[+Dialect <: SqlIdiom, +Naming <: NamingS
     prepare: Prepare = identityPrepare,
     extractor: Extractor[T] = identityExtractor
   )(info: ExecutionInfo, dc: Runner): QCStream[T] = {
-    def prepareStatement(conn: Connection) = {
+    def prepareStatement(conn: Connection): PrepareRow = {
       val stmt         = prepareStatementForStreaming(sql, conn, fetchSize)
       val (params, ps) = prepare(stmt, conn)
       logger.logQuery(sql, params)
       ps
     }
 
-    val managedEnv: ZStream[Connection, Throwable, (Connection, PrepareRow, ResultSet)] =
-      ZStream.scoped {
-        for {
-          conn <- ZIO.service[Connection]
-          ps   <- scopedBestEffort(ZIO.attempt(prepareStatement(conn)))
-          rs   <- scopedBestEffort(ZIO.attempt(ps.executeQuery()))
-        } yield (conn, ps, rs)
-      }
+    val managedEnv: ZIO[Scope with Connection, Throwable, (Connection, ResultSet)] =
+      for {
+        connection         <- scopedBestEffort(ZIO.service[Connection])
+        previousAutoCommit <- ZIO.attempt(connection.getAutoCommit)
+        _ <- ZIO.acquireRelease(ZIO.attempt(connection.setAutoCommit(false))) { _ =>
+               ZIO.debug("setAutoCommit(false) of connection") *>
+                 ZIO
+                   .attempt(connection.setAutoCommit(previousAutoCommit))
+                   .tapError { e =>
+                     ZIO
+                       .attempt(
+                         logger.underlying
+                           .error(s"setAutoCommit(previousAutoCommit) of connection failed", e)
+                       )
+                       .ignore *>
+                       ZIO
+                         .attempt(connection.close())
+                         .tapError(e => ZIO.attempt(logger.underlying.error(s"close() of connection failed", e)).ignore)
+                   }
+                   .orDie
```

Comment:
This is the actually fix fixing the test. If the "release" call of `conn.setAutoCommit` fails, like the test programs it to,  we attempt to close the connection